src/Core/FakeVimCommand.cpp

```diff
@@ -0,0 +1,266 @@
+                appwin->currentWindow()->statusBar()->showMessage(
+                    tr("Error: %1 requires an empty or one of \"cpp\", \"java\" and \"python\" argument").arg(ex.cmd),
+                    STATUS_RESPONSE_TIMEOUT);
+        }
+    }
+
+    return {false, ""};
+}
+
+QPair<bool, QString> FakeVimCommand::wantOpenFile(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "open" || ex.cmd == "opn")
+    {
+        if (ex.args.isEmpty())
+        {
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(tr("Error: %1 requires a file path").arg(ex.cmd),
+                                                                  STATUS_RESPONSE_TIMEOUT);
+            return {true, ""};
+        }
+
+        QFileInfo file(ex.args);
+        if (!file.exists())
+        {
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(tr("Error: %1 does not exists").arg(ex.args),
+                                                                  STATUS_RESPONSE_TIMEOUT);
+            return {true, ""};
+        }
+        else if (!file.isFile())
+        {
+
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(tr("Error: %1 is not a file").arg(ex.args),
+                                                                  STATUS_RESPONSE_TIMEOUT);
+            return {true, ""};
+        }
+
+        else if (!Util::cppSuffix.contains(file.suffix()) && !Util::javaSuffix.contains(file.suffix()) &&
+                 !Util::javaSuffix.contains(file.suffix()))
+        {
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(
+                    tr("Error: %1 is not a source file of C/C++, Java or Python").arg(ex.args),
+                    STATUS_RESPONSE_TIMEOUT);
+
+            return {true, ""};
+        }
+        else
+        {
+            return {true, ex.args};
+        }
+    }
+    return {false, ""};
+}
+
+bool FakeVimCommand::wantCompile(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "compile" || ex.cmd == "cmp";
+}
+
+bool FakeVimCommand::wantCompileRun(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "crun" || ex.cmd == "crn";
+}
+
+bool FakeVimCommand::wantDetachedRun(FakeVim::Internal::ExCommand const &ex)
+{
+    return ex.cmd == "drun" || ex.cmd == "drn";
+}
+
+QPair<bool, int> FakeVimCommand::wantRun(FakeVim::Internal::ExCommand const &ex)
+{
+    if (ex.cmd == "run")
+    {
+        if (ex.args.isEmpty()) // run all testcases
+        {
+            return {true, 0};
+        }
+        bool ok = false;
+        int caseNum = ex.args.toInt(&ok);
+        if (ok)
+            return {true, caseNum};
+        else
+        {
+            if (appwin->currentWindow())
+                appwin->currentWindow()->statusBar()->showMessage(tr("%1 is not a number").arg(caseNum),
```

Comment:
These error messages will timeout after 2 seconds and restore the status bar to its older state.